### PR TITLE
{chem} [intel/2016a] CP2K 3.0 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by tovrstra** [comment]
This is interesting! I'd like to give this a try.

**Comment by tovrstra** [comment]
Most of the dependencies build fine, except that downloading byacc from the ftp URL does not work on the clusters, probably due to the firewall. That's easy enough to work around.

The CP2K eb's fail both with the same error message. I'm not sure how this can be fixed?

```
== temporary log file in case of crash /local/20306.master17.phanpy.gent.vsc/eb-yYda1h/easybuild-pHJ6cq.log
== processing EasyBuild easyconfig ..../CP2K-3.0-intel-2016a-psmp.eb
== building and installing CP2K/3.0-intel-2016a-psmp...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== FAILED: Installation ended unsuccessfully (build directory: /user/data/gent/gvo000/gvo00003/apps_cmm/work/builds/phanpy/CP2K/3.0/intel-2016a-psmp):
Discovering programs ...
Removing stale archives for psmp ...
Resolving dependencies for psmp ...
make[2]: execvp: /bin/sh: Argument list too long
make[2]: *** [makedep] Error 127
make[

```

**Comment by tovrstra** [comment]
This may be related but I don't see how this can be easily incorporated into the eb?
https://www.cp2k.org/faq:libsmm_arguments_too_long

**Comment by wpoely86** [comment]
@tovrstra you probably run too many build process at the same time. Add a `maxparallel` to the easyconfig should do it.

The `psmp` build currently still doesn't work. I've trying to the libxsmm working as it might give a nice speed boost. Any other features on your wish list for 3.0? https://www.cp2k.org/version_history
- Interface to the PEXSI library
- PLUMED 2.0 interface PLUMED (should already be done)
- Interface to ELPA2015 ELPA
- Support for the ASE Python toolkit.

**Comment by tovrstra** [comment]
It would be great if it could already work as you have prepared now. ELPA could be nice at some point but it certainly is not a priority. I'll try the `maxparallel` option.

**Comment by tovrstra** [comment]
So, I tried with `maxparallel = 1` but it didn't help. I figured out what went wrong: the path in which CP2K is too long (111 characters). If I reduce it to 36 characters, with a symbolic link, the problem disappears. I'll have to reorganize my directories.

**Comment by tovrstra** [comment]
I could install `CP2K-3.0-intel-2016a.eb` and run a simple test calculation with it. It seems fine at first sight. I'll also give the psmp version a try. Thanks!

**Comment by wpoely86** [comment]
Yes, the MPI version works but the (x)smm needs explicit testing. The psmp version doesn't even compile. We seem to hit a bug in the intel compiler.

**Comment by tovrstra** [comment]
The test suite of CP2K does not contain explicit tests for the (x)smm but by grepping the source, it seems that it is extensively used in the DBCSR library, for which there are a lot of tests. That may be worth a try. I'll see what I can do.

**Comment by hajgato** [comment]
@wpoely86 
Any progress? I got a `CP2K` request.

**Comment by wpoely86** [comment]
@hajgato the normal version already works, the psmp version gives a compiler crash. 

I will try to flesh out this PR asap.

**Comment by hajgato** [comment]
@wpoely86: I also did it. I also updated the `libxc` to 2.2.3, otherwise is the same than `CP2K-3.0-intel-2016a.eb`. So if you do not have time, do not bother.

**Comment by wpoely86** [comment]
Needs hpcugent/easybuild-easyblocks#942

**Comment by boegel** [comment]
@wpoely86 please sync with `develop` now #3099 is merged

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 7 out of 13 (13 easyconfigs in this PR)
node2665.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/011c644c8849af09c143c31ad7e3f10e for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 10 out of 12 (12 easyconfigs in this PR)
node2624.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ae4fa50ec54139f5d9c2930991513551 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 10 out of 12 (12 easyconfigs in this PR)
node2624.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3988303161c194a9bebdfe3b1b776af7 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 9 out of 12 (12 easyconfigs in this PR)
node2160.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2f803a5f47384da2478340f75fb1ab0c for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 9 out of 12 (12 easyconfigs in this PR)
node2624.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b37aa1ede282efe00ac679be7b1ea453 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 11 out of 13 (13 easyconfigs in this PR)
node2114.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bedcf56d23e1577d6dda9923321e4727 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 11 out of 13 (13 easyconfigs in this PR)
node2716.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ff9e69483cb03cdddedb828a585b9c58 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
node2002.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b2b20e74001a6cfcc13533ecaf14795c for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
node2646.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/95707eb260543efb2e989e1fe6441d4f for a full test report.

**Comment by wpoely86** [comment]
@boegel this one is ready.

File: easybuild/easyconfigs/l/Libint/Libint-1.1.4-intel-2016a.eb [hunk 1]
```diff
@@ -6,7 +6,7 @@ description = """Libint library is used to evaluate the traditional (electron re
  matrix elements (integrals) over Cartesian Gaussian functions used in modern atomic and molecular theory."""
 
 toolchain = {'name': 'intel', 'version': '2016a'}
-toolchainopts = {'pic': True}
+toolchainopts = {'opt': True, 'optarch': True, 'pic': True}
```
Comment (boegel) [inline easybuild/easyconfigs/l/Libint/Libint-1.1.4-intel-2016a.eb]: @wpoely86 what's this about? why add default values?

File: easybuild/easyconfigs/g/GSL/GSL-2.1-intel-2016b.eb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+name = 'GSL'
+version = '2.1'
+
+homepage = 'http://www.gnu.org/software/gsl/'
+description = """The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers.
+ The library provides a wide range of mathematical routines such as random number generators, special functions
+ and least-squares fitting."""
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+toolchainopts = {'opt': True, 'optarch': True, 'unroll': True, 'pic': True}
```
Comment (boegel) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-intel-2016b.eb]: @wpoely86 please drop the `opt` and `optarch`, they're enabled by default

File: easybuild/easyconfigs/l/Libint/Libint-1.1.6-intel-2016b.eb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name = 'Libint'
+version = '1.1.6'
+
+homepage = 'https://sourceforge.net/p/libint/'
+description = """Libint library is used to evaluate the traditional (electron repulsion) and certain novel two-body
+ matrix elements (integrals) over Cartesian Gaussian functions used in modern atomic and molecular theory."""
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+toolchainopts = {'opt': True, 'optarch': True, 'pic': True}
```
Comment (boegel) [inline easybuild/easyconfigs/l/Libint/Libint-1.1.6-intel-2016b.eb]: @wpoely86 no `opt/`optarch` please

File: easybuild/easyconfigs/l/Libint/Libint-1.1.4-intel-2016a.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/l/Libint/Libint-1.1.4-intel-2016a.eb]: make it in sync with the other easyconfig. This one was the exception.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**